app/src/main/java/org/mozilla/fenix/search/SearchComponent.kt

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.search
+
+import android.annotation.SuppressLint
+import android.view.ViewGroup
+import androidx.navigation.Navigation
+import kotlinx.android.synthetic.main.component_search.view.*
+import kotlinx.android.synthetic.main.fragment_browser.*
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.fenix.R
+import org.mozilla.fenix.mvi.Action
+import org.mozilla.fenix.mvi.ActionBusFactory
+import org.mozilla.fenix.mvi.Change
+import org.mozilla.fenix.mvi.Reducer
+import org.mozilla.fenix.mvi.UIComponent
+import org.mozilla.fenix.mvi.ViewState
+
+class SearchComponent(
+    private val container: ViewGroup,
+    override val bus: ActionBusFactory,
+    override var initialState: SearchState = SearchState("")
+) :
+    UIComponent<SearchState, SearchAction, SearchChange>(bus) {
+
+    override val reducer: Reducer<SearchState, SearchChange> = { state, change ->
+        when (change) {
+            is SearchChange.Changed -> state // TODO handle state changes here
+        }
+    }
+
+    override fun initView() = SearchUIView(container, bus)
+    init {
+        setup()
+    }
+
+    fun getView(): BrowserToolbar = uiView.toolbar
+    fun editMode() = getView().editMode()
+
+    @SuppressLint("CheckResult")
+    fun setup(): SearchComponent {
+        render(reducer)
+        getUserInteractionEvents<SearchAction>()
+            .subscribe {
+                Logger("SearchComponent").debug(it.toString())
+                when (it) {
+                    is SearchAction.EditComplete -> {
+                        transitionToBrowser(container)
```

Comment:
Should a UI component know how to navigate?

Reply:
That's a reasonable thing to fix. I can pass a listener as I did in my RB example.